packages/chain-adapters/src/types.ts

```diff
@@ -85,54 +85,28 @@ export type SubscribeTxsInput = {
   accountType?: UtxoAccountType
 }
 
-export type TxFee = {
-  assetId: string
-  value: string
-}
+export type TransferType = unchained.TransferType
+export type TradeType = unchained.TradeType
 
-export type Transaction<T extends ChainId> = {
-  address: string
-  blockHash?: string
-  blockHeight: number
-  blockTime: number
-  chain: T
-  chainId: string
-  confirmations: number
-  txid: string
-  fee?: TxFee
-  status: TxStatus
-  tradeDetails?: TradeDetails
+export type Transaction = Omit<unchained.StandardTx, 'transfers'> & {
   transfers: Array<TxTransfer>
-  data?: TxMetadata
+  data?: unchained.StandardTxMetadata | unchained.evm.TxMetadata | unchained.cosmos.TxMetadata
```

Comment:
`unchained.StandardTxMetadata | unchained.evm.TxMetadata | unchained.cosmos.TxMetadata` could maybe be it's own type.

Reply:
I do agree, and I was going for that initially but I couldn't find a naming that works.
We want to have a similar naming (although namespaced) for `TransactionMetadata` across all domains.

Looking back into it, I guess `TransactionData` works, happy to hear any idea you have for a better naming!

2e15a2b33a8d28a2c98f969d59669d79b2ecddd9

Reply:
It's not perfect, but I can't think of anything better right now!